python/pyarrow/includes/libarrow.pxd

```diff
@@ -2489,6 +2489,8 @@ cdef extern from "arrow/compute/exec/aggregate.h" namespace \
     cdef cppclass CAggregate "arrow::compute::Aggregate":
         c_string function
         shared_ptr[CFunctionOptions] options
+        vector[CFieldRef] target
```

Comment:
I'm confused how we got away with this field never existing.  Perhaps we never use `Aggregate` directly?

Reply:
I was very confused about that as well! The `GroupBy` code called by Python ignored the `FieldRef`s coming from the `Aggregate::target` and mapped the `arguments` to the group-by by position. Now that the code relies on the `.size()` of `Aggregate::target`, we started having issues.